### DISCUSSION TITLE
Consulta Sensores

### DISCUSSION THREAD

**Discussion by matiastito** [dds-jv/foro#423] [TPA - 2024] [answered]
Buenas tardes, espero se ecuentren bien. Los molesto por una consulta sobre el TP2.

Con respecto a los sensores:
1. En los dos sensores, nos dan una interfaz, que es provista por una SDK; siguiendo esta idea, esas interfaces tendran el package de ésa SDK, entonces, si la interfaz la tenemos que crear nosotros, ¿Qué package le creamos?, es como que vamos a crear una interfaz que ya debería existir, es decir, va a tener una package que no va a corresponderse con el de la SDK.
2. En cuanto al **sensor de temperatura**, hace lecturas cada 5minutos, ¿ese 'timmer', lo implementamos nosotros? (usando el método `connect` cada 5minutos para que haga la lectura)
3. O podemos asumir que ya lo hace el sensor, en caso para qué se usa el método `connect`, porque no le encuetro el sentido dado que solo tenemos que **configurar** el sensor implementado `Actions`.
4. En resumen, no me queda claro cómo usamos ese sensor, hasta dónde llega nuestra responsabilidad.

Muchas gracias!

**Comment by flbulgarelli** [comment, marked as answer]
:wave: ¡Hola!

>     1. En los dos sensores, nos dan una interfaz, que es provista por una SDK; siguiendo esta idea, esas interfaces tendran el package de ésa SDK, entonces, si la interfaz la tenemos que crear nosotros, ¿Qué package le creamos?, es como que vamos a crear una interfaz que ya debería existir, es decir, va a tener una package que no va a corresponderse con el de la SDK.

Correcto, deberías definir la interfaz del componente externo en un paquete diferente (en el mundo real, lo instalarías directamente usando `maven`). 

> 
>     2. En cuanto al **sensor de temperatura**, hace lecturas cada 5minutos, ¿ese 'timmer', lo implementamos nosotros? (usando el método `connect` cada 5minutos para que haga la lectura)

Vos no tenés que implementar los sensores de temperatura, sino que tenés que asumir que ya vienen implementados. Será el propio componente con interfaz `TSensor`  el que cada 5 minutos valide las condiciones y ejecute a la `Action` correspondiente. Tu tarea es trabajar contra esa interfaz, y proveer una implementación de `Action` (pero **no** de `TSensor`) que pueda hacer lo que necesites cuando sea ejecutada. 


> 
>     3. O podemos asumir que ya lo hace el sensor, 

Exacto

> en caso para qué se usa el método `connect`, porque no le encuetro el sentido dado que solo tenemos que **configurar** el sensor implementado `Actions

La invocación a `connect` es parte de la configuración del sensor, por lo que tu sistema deberá enviarle ese mensaje.  

En resumen, debés: 

  * Usar `TSensor` (pero no implementarlo, salvo en el contexto de las pruebas unitarias; en tal caso deberías usar un impostor)
  * Proveer una implementación de `Action` que permita la interacción entre tu sistema y el sensor
